src/system_descriptor.jl

```diff
@@ -9,27 +9,24 @@ function SystemArgument(;
     default,
     allowed_values,
 )
-    return SystemArgument(
-        name,
-        default,
-        allowed_values,
-    )
-end
-
-function get_allowed_values(arg::SystemArgument)
-    if isempty(arg.allowed_values)
-        error("allowed values are not defined for $arg")
+    if isempty(allowed_values)
+        error("allowed values cannot be empty")
     else
-        return arg.allowed_values
+        return SystemArgument(
+            name,
+            default,
+            allowed_values,
+        )
     end
 end
 
 get_name(arg::SystemArgument) = arg.name
 get_default(arg::SystemArgument) = arg.default
+get_allowed_values(arg::SystemArgument) = arg.allowed_values
 
-set_allowed_values(arg::SystemArgument, values::Set{<:Any}) = arg.allowed_values = values
 set_name(arg::SystemArgument, name::Symbol) = arg.name = name
 set_default(arg::SystemArgument, default::Any) = arg.default = default
+set_allowed_values(arg::SystemArgument, values::Set{<:Any}) = arg.allowed_values = values
```

Comment:
The struct is now immutable, and so this is not possible. Also, a function that mutates an instance should end with `!`.